src/Yesod/Auth/OAuth2.hs

```diff
@@ -51,3 +61,22 @@ authOAuth2Widget widget name oauth getCreds =
     AuthPlugin name (dispatchAuthRequest name oauth getCreds) login
   where
     login tm = [whamlet|<a href=@{tm $ oauth2Url name}>^{widget}|]
+
+-- | Read from the values set via @'setExtra'@
+getAccessToken :: Creds m -> Maybe AccessToken
```

Comment:
It may be more convenient for users to use `Either String` for all the `get*` functions. For example, that would let users combine them monadically without writing their own conversions between `Either String` and `Maybe`.

Reply:
I don't know that I follow your reasoning: I think we can't really predict what Monad the user will want to be in, so I don't know that `Either` would fit any better at call-sites than `Maybe` (they'll almost certainly be in `Handler` 99% of the time)... But I can get behind `Either` for this because it retains information users could choose to silence when in `Maybe` (e.g. `hush`), vs the current way where the user would have to invent information when in `Either` (e.g. `note`) -- so :+1:

Reply:
Well, on second thought I'm not sure. Having these return `Either` feels like if `Map.lookup` returned either: that'd be silly because there's only one way it can fail.

I think I'm going to keep this for now.